br/pkg/task/stream.go

```diff
@@ -1002,16 +1035,16 @@ func RunStreamTruncate(c context.Context, g glue.Glue, cmdName string, cfg *Stre
 	}
 
 	if cfg.Until < sp {
-		console.Println("According to the log, you have truncated backup data before", em(formatTS(sp)))
+		console.Println("According to the log, you have truncated log backup data before", em(formatTS(sp)))
 		if !cfg.SkipPrompt && !console.PromptBool("Continue? ") {
 			return nil
 		}
 	}
 
-	readMetaDone := console.ShowTask("Reading Metadata... ", glue.WithTimeCost())
+	readMetaDone := console.ShowTask("Reading log backup metadata... ", glue.WithTimeCost())
 	metas := stream.StreamMetadataSet{
 		MetadataDownloadBatchSize: cfg.MetadataDownloadBatchSize,
-		Helper:                    stream.NewMetadataHelper(),
+		Helper:                    stream.NewMetadataHelper(nil),
```

Comment:
if encryption enabled, and truncate doesn't now set encryption manager. will it panic when user try to truncate file?

Reply:
I had the same concern but looking at the code seems what truncate does is read the metafile and figure out if the TS is within the boundary and directly delete the data file based on the path in the metafile without decrypting it. If that's the case it probably won't need the master keys as the metafile is not encrypted right now.

I added an e2e test in the `br_encryption`, it does truncate at the end of the test and did not throw any errors.